fetcher/bin/fetch_gha_min_cost.py

```diff
@@ -0,0 +1,139 @@
+import os
+from typing import Any, Optional, Tuple
+import github
+from datetime import datetime, timezone, timedelta
+import concurrent.futures
+from fetcher.services.github_service import GithubService
+import logging
+
+
+logging.basicConfig(
+    level=logging.INFO, format="%(asctime)s - %(name)s - %(levelname)s - %(message)s"
+)
+logger = logging.getLogger(__name__)
+
+
+def get_environment_variables() -> str:
+    github_app_moj_token = os.getenv("GH_APP_MOJ_TOKEN")
+    if not github_app_moj_token:
+        raise ValueError(
+            "The env variable GH_APP_MOJ_TOKEN is empty or missing")
+    github_app_ap_token = os.getenv("GH_APP_AS_TOKEN")
+    if not github_app_ap_token:
+        raise ValueError(
+            "The env variable GH_APP_AS_TOKEN is empty or missing")
+
+    return github_app_moj_token, github_app_ap_token
+
+
+def calculate_gha_run_minutes_and_cost(run_id: int, repo_object: github.Repository,
+                                       github_service: GithubService, os_multipliers: dict[str, int],
+                                       minute_cost_usd: float) -> Tuple[float, float]:
+
+    total_minutes = 0.0
+    cost_per_run = 0.0
+    response = github_service.get_workflow_run_details(repo_name=repo_object.full_name,
+                                                       run_id=run_id)
+
+    for os_type, multiplier in os_multipliers.items():
+        billable_data = response["billable"].get(os_type)
+        if billable_data:
+            total_ms = billable_data['total_ms']
+            total_minutes += total_ms / 60000
+            cost_per_run += total_minutes * minute_cost_usd * multiplier
+
+    return total_minutes, cost_per_run
+
+
+def calculcate_gha_repo_minutes_and_cost(repo_object: github.Repository, start_date: str, end_date: str,
+                                 github_service: GithubService, os_multipliers: dict[str, int],
+                                 minute_cost_usd: float) -> Tuple[float, float]:
+
```

Comment:
Type hints is great, but I think something to indicate the required  format for the date string would be useful - like maybe a default value like `start_date: str="2024-12-03"` (if that's the right format!), or a regex check for the required format - as presumably it'll error later on if it doesn't get expected format. Unless you think not necessary if only being programmatically passed the date strings.

https://stackoverflow.com/questions/38727520/how-do-i-add-default-parameters-to-functions-when-using-type-hinting

Reply:
Agreed this is only going to be used internally so no need for date format example